diff --git a/group.services.yml b/group.services.yml
index c319dfcbd521e13aeaeece13e345d0c965fd2346..802b35c3098cbbaad483df101447d16c52a60700 100644
--- a/group.services.yml
+++ b/group.services.yml
@@ -1,6 +1,7 @@
 services:
   access_check.group.permission:
     class: 'Drupal\group\Access\GroupPermissionAccessCheck'
+    arguments: ['@group.group_route_context']
     tags:
       - { name: 'access_check', applies_to: '_group_permission' }
   access_check.group.member:
diff --git a/src/Access/GroupPermissionAccessCheck.php b/src/Access/GroupPermissionAccessCheck.php
index bb7901567f175df052c046188cdfdd88797cb38c..25f44e2dabf4889949f2bd78552865a2f781cc3a 100644
--- a/src/Access/GroupPermissionAccessCheck.php
+++ b/src/Access/GroupPermissionAccessCheck.php
@@ -6,7 +6,7 @@ use Drupal\Core\Access\AccessResult;
 use Drupal\Core\Routing\Access\AccessInterface;
 use Drupal\Core\Routing\RouteMatchInterface;
 use Drupal\Core\Session\AccountInterface;
-use Drupal\group\Entity\GroupInterface;
+use Drupal\group\Context\GroupRouteContext;
 use Symfony\Component\Routing\Route;
 
 /**
@@ -14,6 +14,23 @@ use Symfony\Component\Routing\Route;
  */
 class GroupPermissionAccessCheck implements AccessInterface {
 
+  /**
+   * The group route context.
+   *
+   * @var \Drupal\group\Context\GroupRouteContext
+   */
+  protected $groupRouteContext;
+
+  /**
+   * Constructs a new GroupPermissionAccessCheck.
+   *
+   * @param \Drupal\group\Context\GroupRouteContext $group_route_context
+   *   Group context provider.
+   */
+  public function __construct(GroupRouteContext $group_route_context) {
+    $this->groupRouteContext = $group_route_context;
+  }
+
   /**
    * Checks access.
    *
@@ -30,20 +47,9 @@ class GroupPermissionAccessCheck implements AccessInterface {
   public function access(Route $route, RouteMatchInterface $route_match, AccountInterface $account) {
     $permission = $route->getRequirement('_group_permission');
 
-    // Don't interfere if no permission was specified.
-    if ($permission === NULL) {
-      return AccessResult::neutral();
-    }
-
-    // Don't interfere if no group was specified.
-    $parameters = $route_match->getParameters();
-    if (!$parameters->has('group')) {
-      return AccessResult::neutral();
-    }
-
-    // Don't interfere if the group isn't a real group.
-    $group = $parameters->get('group');
-    if (!$group instanceof GroupInterface) {
+    // Don't interfere if no group available.
+    $group = $this->groupRouteContext->getBestCandidate();
+    if (!$group) {
       return AccessResult::neutral();
     }
 
diff --git a/src/Cache/Context/RouteGroupCacheContext.php b/src/Cache/Context/RouteGroupCacheContext.php
index aca3ba13286ba87bc4363463405d2fa471c969c8..8e0eb0ade7d767bb4e2783d125f9dce34ca3e38c 100644
--- a/src/Cache/Context/RouteGroupCacheContext.php
+++ b/src/Cache/Context/RouteGroupCacheContext.php
@@ -46,7 +46,7 @@ class RouteGroupCacheContext implements CacheContextInterface {
    * {@inheritdoc}
    */
   public function getContext() {
-    if ($group = $this->getGroupFromRoute()) {
+    if ($group = $this->getBestCandidate()) {
       // If a group was found on the route, we return its ID as the context.
       if ($gid = $group->id()) {
         return $gid;
diff --git a/src/Context/GroupRouteContext.php b/src/Context/GroupRouteContext.php
index 4fd4b1f3eb1175722d479ec82cdd5930c713aa9d..1586716cf60a021f75bf69c44e9f2054c5ff9fb6 100644
--- a/src/Context/GroupRouteContext.php
+++ b/src/Context/GroupRouteContext.php
@@ -49,7 +49,7 @@ class GroupRouteContext implements ContextProviderInterface {
     $cacheability->setCacheContexts(['route.group']);
 
     // Create a context from the definition and retrieved or created group.
-    $context = new Context($context_definition, $this->getGroupFromRoute());
+    $context = new Context($context_definition, $this->getBestCandidate());
     $context->addCacheableDependency($cacheability);
 
     return ['group' => $context];
diff --git a/src/Context/GroupRouteContextTrait.php b/src/Context/GroupRouteContextTrait.php
index 99e9a41c342cdc3a827418c1db1d7f8472529f33..b23b8b9671b13e1ad12826c0e84bb45903ca511f 100644
--- a/src/Context/GroupRouteContextTrait.php
+++ b/src/Context/GroupRouteContextTrait.php
@@ -2,6 +2,7 @@
 
 namespace Drupal\group\Context;
 
+use Drupal\Core\Entity\EntityInterface;
 use Drupal\group\Entity\GroupInterface;
 
 /**
@@ -69,7 +70,14 @@ trait GroupRouteContextTrait {
     $route_match = $this->getCurrentRouteMatch();
 
     // See if the route has a group parameter and try to retrieve it.
-    if (($group = $route_match->getParameter('group')) && $group instanceof GroupInterface) {
+    $group = $this->getCurrentRouteMatch()->getParameter('group');
+
+    // Regular permissions will have an integer set.
+    if ($group && is_numeric($group)) {
+      return $this->getEntityTypeManager()->getStorage('group')->load($group);
+    }
+    // Group permissions will have an object set.
+    elseif ($group instanceof GroupInterface) {
       return $group;
     }
     // Create a new group to use as context if on the group add form.
@@ -81,4 +89,36 @@ trait GroupRouteContextTrait {
     return NULL;
   }
 
+  /**
+   * Retrieves the group entity from the current route.
+   *
+   * Expands $this->getGroupFromRoute() to also return a Group based if
+   * interacting with Group Content such as nodes assigned to a Group.
+   *
+   * @return \Drupal\group\Entity\GroupInterface|null
+   *   A group entity if one could be found or created, NULL otherwise.
+   */
+  public function getBestCandidate() {
+    if ($group = $this->getGroupFromRoute()) {
+      return $group;
+    }
+
+    if (!empty($this->getCurrentRouteMatch()->getParameters())) {
+      $entities = array_filter(iterator_to_array($this->getCurrentRouteMatch()->getParameters()), static function ($parameter) {
+        return $parameter instanceof EntityInterface;
+      });
+      if (!empty($entities)) {
+        foreach ($entities as $entity) {
+          /** @var \Drupal\group\Entity\Storage\GroupRelationshipStorageInterface $group_relationship_storage */
+          $group_relationship_storage = $this->getEntityTypeManager()->getStorage('group_relationship');
+          foreach ($group_relationship_storage->loadByEntity($entity) as $group_relationship) {
+            return $group_relationship->getGroup();
+          }
+        }
+      }
+    }
+
+    return NULL;
+  }
+
 }
diff --git a/tests/src/Unit/RouteGroupCacheContextTest.php b/tests/src/Unit/RouteGroupCacheContextTest.php
index 52ec24b94fa66140148851b7980e4271455a2629..659c686461d12e0f6cbb7db0f16383327616fa8f 100644
--- a/tests/src/Unit/RouteGroupCacheContextTest.php
+++ b/tests/src/Unit/RouteGroupCacheContextTest.php
@@ -9,7 +9,11 @@ use Drupal\Core\Routing\RouteMatchInterface;
 use Drupal\Tests\UnitTestCase;
 use Drupal\group\Cache\Context\RouteGroupCacheContext;
 use Drupal\group\Entity\GroupInterface;
+use Drupal\group\Entity\GroupRelationship;
 use Drupal\group\Entity\GroupTypeInterface;
+use Drupal\group\Entity\Storage\GroupRelationshipStorageInterface;
+use Drupal\user\Entity\User;
+use Symfony\Component\HttpFoundation\ParameterBag;
 
 /**
  * Tests the route.group cache context.
@@ -49,6 +53,7 @@ class RouteGroupCacheContextTest extends UnitTestCase {
    */
   public function testGetContextNoGroup() {
     $this->currentRouteMatch->getParameter('group')->willReturn(NULL);
+    $this->currentRouteMatch->getParameters()->willReturn(new ParameterBag());
     $this->currentRouteMatch->getRouteName()->willReturn('foo');
 
     $cache_context = new RouteGroupCacheContext($this->currentRouteMatch->reveal(), $this->entityTypeManager->reveal());
@@ -70,6 +75,35 @@ class RouteGroupCacheContextTest extends UnitTestCase {
     $this->assertSame(1, $cache_context->getContext());
   }
 
+  /**
+   * Tests getting context value for group_relationship on the route.
+   *
+   * @covers ::getContext
+   */
+  public function testGetContextWithGroupRelationship() {
+    $this->currentRouteMatch->getParameter('group')->willReturn(NULL);
+    $this->currentRouteMatch->getRouteName()->willReturn('foo');
+
+    $group = $this->prophesize(GroupInterface::class);
+    $group->id()->willReturn(1);
+
+    $user = $this->prophesize(User::class)->reveal();
+
+    $group_relationship = $this->prophesize(GroupRelationship::class);
+    $group_relationship->getGroup()->willReturn($group->reveal());
+    $group_relationship->getEntity()->willReturn($user);
+
+    $storage = $this->prophesize(GroupRelationshipStorageInterface::class);
+    $storage->loadByEntity($user)->willReturn([$group_relationship->reveal()]);
+    $this->entityTypeManager->getStorage('group_relationship')->willReturn($storage->reveal());
+
+    $parameters = new ParameterBag(['user' => $user]);
+    $this->currentRouteMatch->getParameters()->willReturn($parameters);
+
+    $cache_context = new RouteGroupCacheContext($this->currentRouteMatch->reveal(), $this->entityTypeManager->reveal());
+    $this->assertSame(1, $cache_context->getContext());
+  }
+
   /**
    * Tests getting the context value when on the group add form route.
    *
