diff --git a/src/Plugin/CKEditorPlugin/DrupalMediaDecorator.php b/src/Plugin/CKEditorPlugin/DrupalMediaDecorator.php
index e644048..ae571ca 100644
--- a/src/Plugin/CKEditorPlugin/DrupalMediaDecorator.php
+++ b/src/Plugin/CKEditorPlugin/DrupalMediaDecorator.php
@@ -9,7 +9,7 @@ use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
 use Symfony\Component\DependencyInjection\ContainerInterface;
 use Drupal\Core\Plugin\PluginBase;
 use Drupal\editor\Entity\Editor;
-use Drupal\media\Plugin\CKEditorPlugin\DrupalMedia;
+use Drupal\ckeditor\Plugin\CKEditorPlugin\DrupalMedia;
 
 /**
  * Decorates the core drupalmedia plugin.
