diff --git a/sites/example.sites.php b/sites/example.sites.php
index daaf682..686a831 100644
--- a/sites/example.sites.php
+++ b/sites/example.sites.php
@@ -55,3 +55,13 @@
  * @see \Drupal\Core\DrupalKernel::getSitePath()
  * @see https://www.drupal.org/documentation/install/multi-site
  */
+
+/**
+ * Stub for the unl_multisite module to generate site aliases.
+ */
+# THIS SECTION IS AUTOMATICALLY GENERATED
+# DO NOT EDIT!!!!
+
+# %UNL_CREATION_TOOL_STUB%
+
+# END OF AUTOMATICALLY GENERATED AREA
