diff --git a/.htaccess b/.htaccess
index 83cb71a..e906d6e 100644
--- a/.htaccess
+++ b/.htaccess
@@ -80,6 +80,18 @@ AddEncoding gzip svgz
   # even when running as CGI or FastCGI.
   RewriteRule ^ - [E=HTTP_AUTHORIZATION:%{HTTP:Authorization}]
 
+  # START unl_multisite SECTION
+  # Add the following line to your httpd.conf where <DRUPAL_ROOT> is the file system path to the Drupal web root.
+  # RewriteMap drupal_unl_multisite txt:<DRUPAL_ROOT>/.htaccess-subsite-map.txt
+  # Do not uncomment the previous line.
+  RewriteRule .*/cron.php cron.php
+  RewriteRule .*/update.php update.php
+  RewriteRule ^(.*?/(core\/assets|core\/misc|core\/modules|core\/themes|modules|sites|themes))(.*) ${drupal_unl_multisite:$1|$1}$3 [DPI]
+
+  RewriteCond ${drupal_unl_multisite://%{HTTP_HOST}%{REQUEST_URI}|NOT_FOUND} !^NOT_FOUND$
+  RewriteRule (.*) ${drupal_unl_multisite://%{HTTP_HOST}%{REQUEST_URI}|$1} [R,L]
+  # END unl_multisite SECTION
+
   # Block access to "hidden" directories whose names begin with a period. This
   # includes directories used by version control systems such as Subversion or
   # Git to store control files. Files whose names begin with a period, as well
