diff --git a/sites/default/default.settings.php b/sites/default/default.settings.php
index 5922cb1..65a27a9 100644
--- a/sites/default/default.settings.php
+++ b/sites/default/default.settings.php
@@ -763,6 +763,11 @@
  */
 $settings['entity_update_backup'] = TRUE;
 
+/**
+ * Load Herbie profile configuration.
+ */
+include $app_root . '/profiles/herbie/includes/settings.php.inc';
+
 /**
  * Load local development override configuration, if available.
  *
@@ -774,6 +779,6 @@
  * Keep this code block at the end of this file to take full effect.
  */
 #
-# if (file_exists($app_root . '/' . $site_path . '/settings.local.php')) {
-#   include $app_root . '/' . $site_path . '/settings.local.php';
-# }
+if (file_exists($app_root . '/' . $site_path . '/settings.local.php')) {
+  include $app_root . '/' . $site_path . '/settings.local.php';
+}
