diff --git a/core/lib/Drupal/Core/Form/FormBuilder.php b/core/lib/Drupal/Core/Form/FormBuilder.php
index 4d1dc27f71..74c240f34b 100644
--- a/core/lib/Drupal/Core/Form/FormBuilder.php
+++ b/core/lib/Drupal/Core/Form/FormBuilder.php
@@ -796,6 +796,9 @@ public function prepareForm($form_id, &$form, FormStateInterface &$form_state) {
         '#type' => 'hidden',
         '#value' => $form_id,
         '#id' => Html::getUniqueId("edit-$form_id"),
+        '#attributes' => [
+          'data-drupal-selector' => Html::getId("edit-$form_id"),
+        ],
         // Form processing and validation requires this value, so ensure the
         // submitted form value appears literally, regardless of custom #tree
         // and #parents being set elsewhere.
@@ -804,6 +807,8 @@ public function prepareForm($form_id, &$form, FormStateInterface &$form_state) {
     }
     if (!isset($form['#id'])) {
       $form['#id'] = Html::getUniqueId($form_id);
+    }
+    if (!isset($form['#attributes']['data-drupal-selector'])) {
       // Provide a selector usable by JavaScript. As the ID is unique, its not
       // possible to rely on it in JavaScript.
       $form['#attributes']['data-drupal-selector'] = Html::getId($form_id);
@@ -976,11 +981,6 @@ public function doBuildForm($form_id, &$element, FormStateInterface &$form_state
       // possible to rely on it in JavaScript.
       $element['#attributes']['data-drupal-selector'] = Html::getId($unprocessed_id);
     }
-    else {
-      // Provide a selector usable by JavaScript. As the ID is unique, its not
-      // possible to rely on it in JavaScript.
-      $element['#attributes']['data-drupal-selector'] = Html::getId($element['#id']);
-    }
 
     // Add the aria-describedby attribute to associate the form control with its
     // description.
