diff --git a/src/Commands/config/ConfigImportCommands.php b/src/Commands/config/ConfigImportCommands.php
index 2d729e238..f0bff377d 100644
--- a/src/Commands/config/ConfigImportCommands.php
+++ b/src/Commands/config/ConfigImportCommands.php
@@ -175,6 +175,10 @@ class ConfigImportCommands extends DrushCommands
     #[CLI\Topics(topics: [DocsCommands::DEPLOY])]
     public function import(array $options = ['source' => self::REQ, 'partial' => false, 'diff' => false])
     {
+        // UNL project-herbie notice
+        $this->logger()->notice(('You cannot run config:import. Refer to https://github.com/unlcms/project-herbie#configuration-management.'));
+        return;
+
         // Determine source directory.
         $source_storage_dir = ConfigCommands::getDirectory($options['source']);
