diff --git a/external_entities.module b/external_entities.module
index 6b3f417..6f5b22b 100755
--- a/external_entities.module
+++ b/external_entities.module
@@ -8,6 +8,7 @@
 use Drupal\Core\Entity\FieldableEntityInterface;
 use Drupal\Core\Cache\Cache;
 use Drupal\Core\Entity\ContentEntityType;
+use Drupal\Core\Entity\EntityFormInterface;
 use Drupal\Core\Entity\EntityInterface;
 use Drupal\Core\Entity\Display\EntityViewDisplayInterface;
 use Drupal\external_entities\ExternalEntityTypeInterface;
@@ -173,9 +174,14 @@ function external_entities_field_storage_config_edit_form_validate_cardinality(a
  * Implements hook_inline_entity_form_entity_form_alter().
  */
 function external_entities_inline_entity_form_entity_form_alter(&$entity_form, FormStateInterface $form_state) {
+  $form_object = $form_state->getFormObject();
+  if (!$form_object instanceof EntityFormInterface) {
+    return;
+  }
+
   /* @var \Drupal\Core\Entity\FieldableEntityInterface $annotation_entity */
   $annotation_entity = $entity_form['#entity'];
-  $external_entity = $form_state->getFormObject()->getEntity();
+  $external_entity = $form_object->getEntity();
   if ($annotation_entity->isNew() && $external_entity instanceof ExternalEntityInterface) {
     $external_entity_type = $external_entity->getExternalEntityType();
     if ($external_entity_type->isAnnotatable()
