diff --git a/src/ExternalEntityStorage.php b/src/ExternalEntityStorage.php
index ed8a536..bebe8de 100755
--- a/src/ExternalEntityStorage.php
+++ b/src/ExternalEntityStorage.php
@@ -210,11 +210,11 @@ class ExternalEntityStorage extends ContentEntityStorageBase implements External
   protected function getFromExternalStorage(array $ids) {
     $entities = [];
 
-    if (!empty($ids)) {
-      // Sanitize IDs. Before feeding ID array into buildQuery, check whether
-      // it is empty as this would load all entities.
-      $ids = $this->cleanIds($ids);
-    }
+//    if (!empty($ids)) {
+//      // Sanitize IDs. Before feeding ID array into buildQuery, check whether
+//      // it is empty as this would load all entities.
+//      $ids = $this->cleanIds($ids);
+//    }
 
     if ($ids) {
       $data = $this
