diff --git a/src/PermissionManager.php b/src/PermissionManager.php
index a0453e1..de24df0 100644
--- a/src/PermissionManager.php
+++ b/src/PermissionManager.php
@@ -101,7 +101,7 @@ class PermissionManager implements PermissionManagerInterface {
         $id = $this->getPermissionMachineNameFromKey($removed_permission);
         $perm_entity = $storage->load($id);
 
-        if ($perm_entity->permissionHasRole($role_id)) {
+        if (isset($perm_entity) && $perm_entity->permissionHasRole($role_id)) {
           $perm_entity->removeRoleFromPermission($role_id);
 
           // If permission is no longer assigned to any roles, then delete
