diff --git a/src/Plugin/Linkit/Matcher/EntityMatcher.php b/src/Plugin/Linkit/Matcher/EntityMatcher.php
index 9d56a47..03d9605 100644
--- a/src/Plugin/Linkit/Matcher/EntityMatcher.php
+++ b/src/Plugin/Linkit/Matcher/EntityMatcher.php
@@ -491,7 +491,18 @@ class EntityMatcher extends ConfigurableMatcherBase {
    *   The path for this entity.
    */
   protected function buildPath(EntityInterface $entity) {
-    return $entity->toUrl('canonical', ['path_processing' => FALSE])->toString();
+    $path = $entity->toUrl('canonical', ['path_processing' => FALSE])->toString();
+    // For media entities, check if standalone URLs are allowed. If not, then
+    // strip '/edit' from the end of the canonical URL returned
+    // by $entity->toUrl().
+    if ($entity->getEntityTypeId() == 'media') {
+      $standalone_url = \Drupal::config('media.settings')->get('standalone_url');
+      if (!$standalone_url) {
+        // Strip "/edit".
+        $path = substr($path, 0, -5);
+      }
+    }
+    return $path;
   }
 
   /**
