diff --git a/core/modules/mysql/src/Driver/Database/mysql/Connection.php b/core/modules/mysql/src/Driver/Database/mysql/Connection.php
index 246da85af0..ada593851b 100644
--- a/core/modules/mysql/src/Driver/Database/mysql/Connection.php
+++ b/core/modules/mysql/src/Driver/Database/mysql/Connection.php
@@ -409,7 +409,13 @@ protected function popCommittableTransactions() {
       else {
         // Attempt to release this savepoint in the standard way.
         try {
-          $this->query('RELEASE SAVEPOINT ' . $name);
+
+          // UNL CHANGE: https://github.com/unlcms/project-herbie/issues/366
+          $is_herbie_install = \Drupal::state()->get('is_herbie_install');
+          if (!$is_herbie_install) {
+            $this->query('RELEASE SAVEPOINT ' . $name);
+          }
+
         }
         catch (DatabaseExceptionWrapper $e) {
           // However, in MySQL (InnoDB), savepoints are automatically committed
@@ -419,6 +425,13 @@ protected function popCommittableTransactions() {
           //
           // To avoid exceptions when no actual error has occurred, we silently
           // succeed for MySQL error code 1305 ("SAVEPOINT does not exist").
+
+          // UNL CHANGE: https://github.com/unlcms/project-herbie/issues/366
+          \Drupal::logger('herbie')->debug('RELEASE SAVEPOINT %name failed.',
+            array(
+              '%name' => $name,
+            ));
+
           if ($e->getPrevious()->errorInfo[1] == '1305') {
             // If one SAVEPOINT was released automatically, then all were.
             // Therefore, clean the transaction stack.
