diff --git a/src/Plugin/Filter/MediaEmbedDecorator.php b/src/Plugin/Filter/MediaEmbedDecorator.php
index 75995c0..9ec0188 100644
--- a/src/Plugin/Filter/MediaEmbedDecorator.php
+++ b/src/Plugin/Filter/MediaEmbedDecorator.php
@@ -170,10 +170,14 @@ final class MediaEmbedDecorator extends FilterBase implements ContainerFactoryPl
     // by 'allowed_media_types'. If no bundles are listed in
     // 'allowed_media_types', then all bundles are allowed and all
     // configuration should be saved.
-    if (!empty($configuration['settings']['allowed_media_types'])) {
-      $configuration['settings']['bundle_view_modes'] = array_intersect_key($configuration['settings']['bundle_view_modes'], $configuration['settings']['allowed_media_types']);
+    if (!empty($configuration['settings']['allowed_media_types'])
+      && !empty($configuration['settings']['bundle_view_modes'])
+      ) {
+      $configuration['settings']['bundle_view_modes'] = array_intersect_key(
+        $configuration['settings']['bundle_view_modes'],
+        $configuration['settings']['allowed_media_types']
+      );
     }
-
     parent::setConfiguration($configuration);
   }
 
